enhancements/profile-versioning.md

```diff
@@ -0,0 +1,449 @@
+ocp4-cis-node-1.5.0              22s   1.5.0
+ocp4-e8                          21s
+ocp4-pci-dss                     21s   3.2.1
+ocp4-pci-dss-node                21s   3.2.1
+rhcos4-e8                        11s
+```
+
+This approach gives user the ability to roll forward with new versions of a
+profile, while providing the flexibility to pin to specific versions of release
+profiles.
+
+From a [ComplianceAsCode/content](https://github.com/ComplianceAsCode/content)
+perspective, the profiles are completely separate.
+
+Users should be able to create a `ScanSettingBinding` without specifying a version:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSettingBinding
+metadata:
+  name: cis
+  namespace: openshift-compliance
+profiles:
+  - name: ocp4-cis
+    kind: Profile
+    apiGroup: compliance.openshift.io/v1alpha1
+  - name: ocp4-cis-node
+    kind: Profile
+    apiGroup: compliance.openshift.io/v1alpha1
+settingsRef:
+  name: default
+  kind: ScanSetting
+  apiGroup: compliance.openshift.io/v1alpha1
+```
+
+This will result in the scan using the latest version of a benchmark. This is
+the default behavior that the Compliance Operator exposes today, where updates
+to profiles are always rolled forward. This behavior must be explicitly
+documented, even though it was the default prior to supporting multiple
+versions.
+
+However, users have the ability to reference a specific scan name to scan using
+an older version of the profile.
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSettingBinding
+metadata:
+  name: cis
+  namespace: openshift-compliance
+profiles:
+  - name: ocp4-cis-1.4.0
+    kind: Profile
+    apiGroup: compliance.openshift.io/v1alpha1
+  - name: ocp4-cis-node-1.4.0
+    kind: Profile
+    apiGroup: compliance.openshift.io/v1alpha1
+settingsRef:
+  name: default
+  kind: ScanSetting
+  apiGroup: compliance.openshift.io/v1alpha1
+```
+
+The above binding would only run scans using the `ocp4-cis-1.4.0` profile.
+
+#### Immutability
+
+By default, the `version` attribute, label, or annotation of a `Profile` should
+be owned by the operator, which is responsible for parsing the data stream. At
+this point, we're not going to support users setting, or overriding, the
+profile version.
+
+### Risks and Mitigations
+
+Contributors updating profile content will need to be mindful of the profile
+version they're implementing. Failure to update the version of a profile will
+result in new content being delivered in an version. While this isn't riskier
+than anything we do today, it will be misleading for users.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. What happens when a `Profile` extended by a `TailoredProfile` changes?
+
+Users can now extend an older version of a profile, but this profile won't
+receive updates or new features. Users extending a versionless profile name
+will always be extending the latest version, which is how `TailoredProfiles`
+work today.
```

Comment:
another thing is I think we need to find out what to do is when the user chooses to extend on an older version of the profile, and on the next update, we do not support such version anymore.